### PR TITLE
[PRM-830] - Updated compliance view + logic for new getObligationAPI

### PR DESCRIPTION
**Pull request by LewisMitchell**
Monthly (18 months history + 6 months for compliance):
monthly_filer_with_24_mths_history
Monthly (15 months history + 6 months for compliance):
monthly_filer_21_mths_history
Quarterly filer:
quarterly_filer

### CODE REVIEW

**Review by ruthdas03 — COMMENTED** [review]
The changes looks ok. Just few observations to consider.

File: app/services/ComplianceService.scala [hunk 1]
```diff
@@ -17,14 +17,69 @@
...
+        val fromDate: LocalDate = fromDateTime.toLocalDate
+        val filingFrequency: FilingFrequencyEnum.Value = getFilingFrequency(pointsThresholdAsString)
+        val toDate: LocalDate = getToDateFromFilingFrequency(lspCreationDate, filingFrequency)
+        connector.getComplianceDataFromDES(vrn, fromDate, toDate).map {
+          complianceData => {
+            val amountOfFulfilledSubmissionsInPast2Years: Int = complianceData.obligationDetails.count(obligation =>
+              obligation.status == ComplianceStatusEnum.fulfilled && obligation.inboundCorrespondenceToDate.isBefore(LocalDate.now()))
+            logger.debug(s"[ComplianceService][getDESComplianceData] - Amount of fulfilled submissions in " +
+              s"the past 2 years: $amountOfFulfilledSubmissionsInPast2Years")
+            val submissionsForNeededForMonthlyFiler: Int = (amountOfFulfilledSubmissionsInPast2Years + 6) - 24
```
[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/services/ComplianceService.scala:48]: Good spot will do!

File: app/viewmodels/TimelineHelper.scala [hunk 2]
```diff
@@ -16,40 +16,60 @@
...
-import javax.inject.Inject
-import models.User
-import models.compliance.Return
+import models.{FilingFrequencyEnum, User}
+import models.compliance.{ComplianceData, CompliancePayload, ComplianceStatusEnum, ObligationDetail}
 import play.api.i18n.Messages
 import play.twirl.api.Html
 import utils.MessageRenderer.getMessage
-import utils.{ImplicitDateFormatter, ViewUtils}
+import utils.{ImplicitDateFormatter, SessionKeys, ViewUtils}
```
[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/viewmodels/TimelineHelper.scala:24]: I noticed this as part of my cleanup task, but I'll pick it up now

File: app/viewmodels/CompliancePageHelper.scala [hunk 3]
```diff
@@ -16,30 +16,38 @@
 
 package viewmodels
 
-import javax.inject.Inject
-import models.compliance.MissingReturn
+import models.User
```
[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/viewmodels/CompliancePageHelper.scala:19]: I noticed this as part of my cleanup task, but I'll pick it up now

**Review by ruthdas03 — APPROVED** [review]

**Pull request merged**